src/components/Account/useAccountCreate.tsx

```diff
@@ -0,0 +1,32 @@
+import { useClient } from '@vocdoni/react-providers'
+import { useMutation, UseMutationOptions, useQuery } from '@tanstack/react-query'
+import { Account } from '@vocdoni/sdk'
+import { useAccountHealthTools } from '~components/Account/use-account-health-tools'
+
+export interface CreateAccountParams {
+  name: string
+  description: string
+}
+
+export const useAccountCreate = (
+  options?: Omit<UseMutationOptions<void, Error, CreateAccountParams>, 'mutationFn'>
+) => {
+  const { createAccount, updateAccount } = useClient()
+  // we want to know if account exists, not the organization (slight difference)
+  const { exists } = useAccountHealthTools()
+
+  const create = async (values: CreateAccountParams) => {
+    let call = () =>
+      createAccount({
+        account: new Account(values),
+      })
+
+    if (exists) {
+      call = () => updateAccount(new Account(values))
```

Comment:
@selankon why is here the account updating implicit? Given that is a transaction that costs should it be done only when directly asked by the user?

Reply:
I am not sure... This code is inherited from the old scaffold... I just moved the logic into a mutation to share it with the saas logic.  Take a look here https://github.com/vocdoni/ui-scaffold/blob/develop/src/components/Account/Create.tsx#L68-L79

Maybe this is an question for ElBoletaire

Reply:
It's not an implicit call. That `call` method is called later when required